verification/spec/connection-handshake/ch.tla

```diff
@@ -0,0 +1,188 @@
+getConnectioEnd(targetParty) ==
+    IF targetParty = "A"
+    THEN 
+        [   state |-> "UNINIT",
+            connectionID |-> "connAtoB",
+            clientID |-> "clientA",
+            counterpartyConnectionID |-> "connBtoA",
+            counterpartyClientID |-> "clientB" ]
+    ELSE IF targetParty = "B"
+         THEN
+            [   state |-> "UNINIT",
+                connectionID |-> "connBtoA",
+                clientID |-> "clientB",
+                counterpartyConnectionID |-> "connAtoB",
+                counterpartyClientID |-> "clientA" ]
+         ELSE Null    
+
+
+\* Constructs a 'Init' Datagram.
+\* Fields are hardcoded for the moment
+getInitDatagram(targetParty) ==
+    [ type |-> "ConnOpenInit", 
+      connectionEnd |-> getConnectioEnd(targetParty) ]
+
+
+isPartyInState(p, state) ==
+    parties[p].connectionEnd.state = state
+
+
+\* Would be ideal to import the following function.
+\* Cf. ibc-rs#55 (file: ConnectionHandlers.tla)
+getConnectionID(p) ==
+    IF p = "A"
+    THEN "connAtoB"
+    ELSE IF p = "B"
+         THEN "connBtoA"
+         ELSE Null  
+
+
+(********************************** Environment functions
+ ***************************************************************************)
+
+    
+\* The environment generates the ConnOpenInit datagram, sending it to 'party'
+GenerateConnOpenInit(targetParty) ==
+    LET initDatagram == getInitDatagram(targetParty)
+    IN inboundDatagrams' = [
+                                inboundDatagrams EXCEPT
+                                ![targetParty] = inboundDatagrams[targetParty] \union { initDatagram } 
+                           ]
+    /\ UNCHANGED <<parties>>
+
+
+EnvStep ==
+    \E targetParty \in Parties :
+        GenerateConnOpenInit(targetParty)
+
+
+(**************************** Party functions
+ ***************************************************************************)
+
+ConnectionEndInit ==
+    [state |-> "UNINIT",
+     connectionID |-> Null,
+     clientID |-> Null,
+     counterpartyConnectionID |-> Null,
+     counterpartyClientID |-> Null ] 
+
+
+(**
+ * Connection Handshake Module handlers
+ **)
+
+CHModuleHandleInit(party) ==
+    /\ isPartyInState(party, "UNINIT")
+    /\  LET connOpenInitDgrs == {dgr \in inboundDatagrams[party] : 
+                                /\ dgr.type = "ConnOpenInit"
+                                /\ dgr.connectionID = getConnectionID(party)}
+        IN IF connOpenInitDgrs /= {} 
+           THEN LET connOpenInitDgr == CHOOSE dgr \in connOpenInitDgrs : TRUE 
+                IN LET connOpenInitConnectionEnd == [
+                     state |-> "INIT",
+                     connectionID |-> connOpenInitDgr.connectionID,
+                     clientID |-> connOpenInitDgr.clientID,
+                     counterpartyConnectionID |-> connOpenInitDgr.counterpartyConnectionID,
+                     counterpartyClientID |-> connOpenInitDgr.counterpartyClientID ] 
+                  IN parties' = [ parties EXCEPT
```

Comment:
It is good practice to return a value from a handler, for example updated state and messages that should be sent. And then you have multiple handlers for multiple message types, and at one place (for example ModuleStep) you just call a given handler and set variables (with ') with a return values.